### PR TITLE
net-p2p/deluge: fix USE-conditional installation of UI files

### PR DESCRIPTION
**Pull request by PPed72**
Gentoo-bug: 381837
Fix patch file for EAPI=6
Bump ebuild to EAPI=6
Add description for new USE flag.
Remove old
Sync live ebuild

### CODE REVIEW

**Comment by monsieurp** [comment]
@gentoo/proxy-maint

File: net-p2p/deluge/deluge-9999.ebuild [hunk 1]
```diff
@@ -44,14 +55,26 @@ RDEPEND=">=net-libs/rb_libtorrent-0.14.9[python]
...
+		-e 's|"new_release_check": True|"new_release_check": False|'
+		-e 's|"check_new_releases": True|"check_new_releases": False|'
+		-e 's|"show_new_releases": True|"show_new_releases": False|'
+	)
+	sed -i "${args[@]}" -- 'deluge/core/preferencesmanager.py' || die
+
+	loc_dir="${S}/deluge/i18n"
+	l10n_find_plocales_changes "${loc_dir}" "" ".po"
+	rm_loc() {
+		rm -vf "${loc_dir}/${1}.po"
```
Comment (monsieurp) [inline net-p2p/deluge/deluge-9999.ebuild]: Missing `|| die`. Actually all your `rm`s have missing `die`s.

File: net-p2p/deluge/deluge-1.3.12-r4.ebuild [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		-e 's|"new_release_check": True|"new_release_check": False|'
+		-e 's|"check_new_releases": True|"check_new_releases": False|'
+		-e 's|"show_new_releases": True|"show_new_releases": False|'
+	)
+	sed -i "${args[@]}" -- 'deluge/core/preferencesmanager.py' || die
+
+	loc_dir="${S}/deluge/i18n"
+	l10n_find_plocales_changes "${loc_dir}" "" ".po"
+	rm_loc() {
+		rm -vf "${loc_dir}/${1}.po"
```
Comment (monsieurp) [inline net-p2p/deluge/deluge-1.3.12-r4.ebuild]: Ditto in this ebuild.

File: net-p2p/deluge/deluge-1.3.12-r4.ebuild [hunk 3]
```diff
@@ -44,14 +55,26 @@ RDEPEND=">=net-libs/rb_libtorrent-0.14.9[python]
...
 	)
+	sed -i "${args[@]}" -- 'setup.py' || die
+	args=(
+		-e 's|"new_release_check": True|"new_release_check": False|'
+		-e 's|"check_new_releases": True|"check_new_releases": False|'
+		-e 's|"show_new_releases": True|"show_new_releases": False|'
+	)
+	sed -i "${args[@]}" -- 'deluge/core/preferencesmanager.py' || die
+
+	loc_dir="${S}/deluge/i18n"
```
Comment (rindeal) [inline net-p2p/deluge/deluge-1.3.12-r4.ebuild]: should be `local`

File: net-p2p/deluge/deluge-1.3.12-r4.ebuild [hunk 4]
```diff
@@ -24,18 +25,28 @@ fi
...
+	sound? ( gtk )
+	libnotify? ( gtk )
+"
+PATCHES=(
+	"${FILESDIR}/${PN}-1.3.12-fix_scheduler_plugin.patch"
+	"${FILESDIR}/${PN}-1.3.5-disable_libtorrent_internal_copy.patch"
+)
 
-DEPEND=">=net-libs/rb_libtorrent-0.14.9[python]
+CDEPEND=">=net-libs/rb_libtorrent-0.14.9[python]"
```
Comment (rindeal) [inline net-p2p/deluge/deluge-1.3.12-r4.ebuild]: is missing `${PYTHON_USEDEP}`

File: net-p2p/deluge/deluge-1.3.12-r4.ebuild (hunk 3 above)
Comment (PPed72) [inline net-p2p/deluge/deluge-1.3.12-r4.ebuild]: In data sabato 23 aprile 2016 03:17:35 CEST, Jan Chren ha scritto:
> >  	)
> > 
> > +	sed -i "${args[@]}" -- 'setup.py' || die
> > +	args=(
> > +		-e 's|"new_release_check": True|"new_release_check": False|'
> > +		-e 's|"check_new_releases": True|"check_new_releases": False|'
> > +		-e 's|"show_new_releases": True|"show_new_releases": False|'
> > +	)
> > +	sed -i "${args[@]}" -- 'deluge/core/preferencesmanager.py' || die
> > +
> > +	loc_dir="${S}/deluge/i18n"
> 
> should be `local`

Do you mean "args" or "loc_dir"? "args" has already been declared local a few 
lines before.

-- 
Paolo Pedroni
paolo.pedroni<at>iol.it

File: net-p2p/deluge/deluge-9999.ebuild [hunk 5]
```diff
@@ -24,18 +25,27 @@ fi
...
+	${PYTHON_REQUIRED_USE}
+	sound? ( gtk )
+	libnotify? ( gtk )
+"
+PATCHES=(
+	"${FILESDIR}/${PN}-1.3.12-fix_scheduler_plugin.patch"
+)
 
-DEPEND=">=net-libs/rb_libtorrent-0.14.9[python]
+CDEPEND=">=net-libs/rb_libtorrent-0.14.9[${PYTHON_USEDEP}]"
```
Comment (rindeal) [inline net-p2p/deluge/deluge-9999.ebuild]: now it's missing `python`, it has to have both as they mean different things

**Comment by PPed72** [comment]
Is this good to go, now?

**Comment by monsieurp** [comment]
@krifisk

**Comment by krifisk** [comment]
Hi,

Please rebase the changes into logical blocks to be added to upstream repo, removing old is fine on its own but the rest should be squashed into one commit (might also get quicker response if opening a bug in bugs.gentoo.org and providing a unified diff there instead of using github)

If console use flag is default UI, should it default to on and should it be added a required use for at least one of the UI providers to be present?

Otherwise looks good for me after a quick walkthrough (cudos for shebang change to init script..)

**Comment by PPed72** [comment]
I will do ASAP, as time permits.

It is not necessary for one of the UI to be present. One can also just install the daemon and control it from another PC (that's what I do, actually)

**Comment by PPed72** [comment]
@krifisk I have added you to gentoo bug #381837, where I posted the unified diff we were talking about.

**Comment by PPed72** [comment]
Ok. I'm closing this PR and will follow up with patches on gentoo bug #381837.

**Pull request closed without merging**